### PR TITLE
Registered users list and change the  status of user

### PR DESCRIPTION
**Pull request by Shibon-ll**
### Requirements

Api to get the registered users list
Api to change the status of user

### Description of the Change
Added GET request to get  the users list from the Amazon Cognito user pool.
Added PUT request by id to change the status of the specified user by administrator
Added unit testcase

### Benefits
Admin can able to see the list of registered users
Admin can able to change the status of the user as Enable/Disable

### Possible Drawbacks
Nil

### Applicable Issues

Nil

### CODE REVIEW

File: core/jazz_usermanagement/test/test.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ describe('User Management', function () {
 
 	it("should throw a invalid or missing arguments for undefined method", function () {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/test/test.js]: Add test case for valid scenario

File: core/jazz_usermanagement/test/test.js [hunk 2]
```diff
@@ -149,6 +149,43 @@ describe('User Management', function () {
 			index.handler(event, context, callback).includes("101");
 		assert.isTrue(bool);
 	});
+
+	it('should throw error validateUserListQuery', function () {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/test/test.js]: Add corresponding success scenarios

File: core/jazz_usermanagement/test/test.js [hunk 3]
```diff
@@ -486,4 +523,136 @@ describe("inside index handler else condition", function () {
...
+			if (err) {
+				return err;
+			} else {
+				return JSON.stringify(responseObj);
+			}
+		};
+
+	});
+
+	it('Should return userList ', function () {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/test/test.js]: Test case name should be specific, meaningful, and can understand the context

File: core/jazz_usermanagement/test/test.js [hunk 4]
```diff
@@ -486,4 +523,136 @@ describe("inside index handler else condition", function () {
...
+		index.handler(event, context, (err, res) => {
+			expect(res.data.result).to.eq('Success');
+			sinon.assert.calledOnce(validateUserListQuery);
+			sinon.assert.calledOnce(getUserList);
+			validateUserListQuery.restore();
+			getUserList.restore();
+		});
+	});
+
+	it('Should return error', function () {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/test/test.js]: Test case name should be specific, meaningful, and can understand the context

File: core/jazz_usermanagement/test/test.js [hunk 5]
```diff
@@ -486,4 +523,136 @@ describe("inside index handler else condition", function () {
...
+		index.handler(event, context, (err, res) => {
+			expect(JSON.parse(err).errorCode).to.eq("500");
+			sinon.assert.calledOnce(validateUserListQuery);
+			sinon.assert.notCalled(getUserList);
+			validateUserListQuery.restore();
+			getUserList.restore();
+		});
+	});
+
+	it('Should return validation error', function () {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/test/test.js]: Test case name should be specific, meaningful, and can understand the context

File: core/jazz_usermanagement/index.js [hunk 6]
```diff
@@ -61,18 +61,18 @@ function handler(event, context, cb)  {
...
 			logger.info('User password reset Request:' + JSON.stringify(service_data));
 
 			exportable.validateResetParams(service_data)
-			.then(() => exportable.forgotPassword(cognito, config, service_data) )
+				.then(() => exportable.forgotPassword(cognito, config, service_data))
 				.then(result => {
-						  logger.info("Password reset was successful for user: " + service_data.email);
-						  return cb(null, responseObj({ result: "success", errorCode: "0", message: "Password reset was successful for user: " + service_data.email }));
-					  })
+					logger.info("Password reset was successful for user: " + service_data.email);
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: logger.debug

File: core/jazz_usermanagement/index.js [hunk 7]
```diff
@@ -61,18 +61,18 @@ function handler(event, context, cb)  {
 		var subPath = getSubPath(event.resourcePath);
 		const cognito = new AWS.CognitoIdentityServiceProvider({ apiVersion: '2016-04-19', region: config.REGION });
 
-		if (subPath.indexOf('reset') > -1) {
+		if (subPath.indexOf('reset') > -1 && event.method === 'POST') {
 			logger.info('User password reset Request:' + JSON.stringify(service_data));
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: logger.debug

File: core/jazz_usermanagement/index.js [hunk 8]
```diff
@@ -83,11 +83,11 @@ function handler(event, context, cb)  {
 						return cb(JSON.stringify(errorHandler.throwInternalServerError("106", "Failed while resetting user password for: " + service_data.email)));
 					}
 				});
-		} else if (subPath.indexOf('updatepwd') > -1) {
+		} else if (subPath.indexOf('updatepwd') > -1 && event.method === 'POST') {
 			logger.info('User password update Request::' + JSON.stringify(service_data));
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: Change logger.info to logger.debug here and other places

File: core/jazz_usermanagement/index.js [hunk 9]
```diff
@@ -83,11 +83,11 @@ function handler(event, context, cb)  {
...
 
 			exportable.validateUpdatePasswordParams(service_data)
-				.then(() =>  exportable.updatePassword(cognito, config, service_data))
+				.then(() => exportable.updatePassword(cognito, config, service_data))
 				.then(result => {
 					logger.info("Successfully updated password for user: " + service_data.email);
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: same as above

File: core/jazz_usermanagement/index.js [hunk 10]
```diff
@@ -106,7 +106,7 @@ function handler(event, context, cb)  {
 						return cb(JSON.stringify(errorHandler.throwInternalServerError("106", "Failed while updating user password for: " + service_data.email)));
 					}
 				});
-		} else {
+		} else if (event.method === 'POST') {
 			logger.info('User Reg Request::' + JSON.stringify(service_data));
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: same as above

File: core/jazz_usermanagement/index.js [hunk 11]
```diff
@@ -131,6 +131,57 @@ function handler(event, context, cb)  {
 					}
 				});
 		}
+
+		if (event.method === 'GET') {
+			var queryParams = event.query;
+			exportable.validateUserListQuery(queryParams, config)
+				.then((params) => exportable.getUserList(cognito, params))
+				.then(results => {
+					logger.info(results);
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: same as above with valid message

File: core/jazz_usermanagement/index.js [hunk 12]
```diff
@@ -131,6 +131,57 @@ function handler(event, context, cb)  {
...
+					return cb(null, responseObj({ result: 'Success', users: results.Users, paginationtoken: results.PaginationToken }, { queryParams }));
+				})
+				.catch(function (err) {
+					logger.error("Failed while Fetching user list" + JSON.stringify(err));
+					if (err.errorType) {
+
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: Remove unwanted space

File: core/jazz_usermanagement/index.js [hunk 13]
```diff
@@ -131,6 +131,57 @@ function handler(event, context, cb)  {
...
+						return cb(JSON.stringify(err));
+					} else {
+						if (err.code) {
```
Comment (SiniWilson) [inline core/jazz_usermanagement/index.js]: Remove unwanted space from here and other places

**Comment by supritat** [comment]
@Shibon-ll, Please work on above mentioned comments and resolve the conflicts.

**Pull request closed without merging**